command/flag.go

```diff
@@ -41,3 +44,123 @@ func (e EnumValue) String() string {
 func (e EnumValue) Get() interface{} {
 	return e
 }
+
+type MapValue struct {
+	m map[string]string
+}
```

Comment:
We can use something like
```go
type MapValue map[string]string

func (m MapValue) String() string
func (m MapValue) Set(s string) error
func (m MapValue) Get() any
```

`map` can be allocated lazily in `Set` method;
```go
func (m MapValue) Set(s string) error {
	// checks etc.

	if m == nil {
		m = make(map[string]string)
	}

	// set logic and stuff...
}
```

and in `MapFlag` you can just;
```go
if f.Value == nil {
	f.Value = MapValue{}
}
```

Reply:
Yeah, seems better. I have refactored. [a97c889](https://github.com/peak/s5cmd/pull/621/commits/a97c889e2ccd3cc09d11744d702bc9b5a3b5d4bf)